compiler/src/org.graalvm.compiler.nodes/src/org/graalvm/compiler/nodes/AbstractBeginNode.java

```diff
@@ -116,6 +118,15 @@ public void generate(NodeLIRBuilderTool gen) {
         };
     }
 
+    public List<Node> getBlockNodeList() {
+        StructuredGraph.ScheduleResult schedule = graph().getLastSchedule();
+        if (schedule == null) {
+            return null;
+        }
+        Block block = schedule.getCFG().blockFor(this);
```

Comment:
You need a couple more guards in there to be safe.  Something like this:
`        if (schedule.getCFG().getNodeToBlock().isNew(this)) {
            // This can occur when nodes were created after the last schedule
            return null;
        }
        Block block = schedule.getCFG().blockFor(this);
        if (block == null) {
            return null;
        }
        return schedule.nodesFor(block);
`
I think we should move this over to IfNode as well.  It's not really something we want to advertise for general use.